src/Cms/PluginAsset.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+namespace Kirby\Cms;
+
+use Kirby\Filesystem\F;
+
+/**
+ * Representing a plugin asset with methods
+ * to manage the asset file between the plugin
+ * and media folder
+ *
+ * @package   Kirby Cms
+ * @author    Nico Hoffmann <nico@getkirby.com>
+ * @link      https://getkirby.com
+ * @copyright Bastian Allgeier
+ * @license   https://getkirby.com/license
+ */
+class PluginAsset
+{
+	public function __construct(
+		protected string $path,
+		protected string $root,
+		protected Plugin $plugin
+	) {
+	}
+
+	public function extension(): string
+	{
+		return F::extension($this->path());
+	}
+
+	public function filename(): string
+	{
+		return F::filename($this->path());
+	}
+
+	/**
+	 * Create a unique media hash
+	 */
+	public function mediaHash(): string
+	{
+		return crc32($this->filename()) . '-' . $this->modified();
+	}
+
+	/**
+	 * Absolute path to the asset file in the media folder
+	 */
+	public function mediaRoot(): string
+	{
+		return $this->plugin()->mediaRoot() . '/' . $this->path();
+	}
+
+	/**
+	 * Public accessible url path for the asset
+	 */
+	public function mediaUrl(): string
+	{
+		return $this->plugin()->mediaUrl() . '/' . $this->path() . '?m=' . $this->mediaHash();
+	}
+
+	/**
+	 * Timestamp when asset file was last modified
+	 *
+	 * @codeCoverageIgnore
```

Comment:
We can test this with `touch($file, 1337000000)` in the test setup to get a known modification time.